discovery-infra/tests/test_wrong_boot_order.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+
+from tests.base_test import BaseTest
+
+
+class TestWrongBootOrder(BaseTest):
+    @pytest.mark.regression
+    def test_wrong_boot_order_one_node(self, api_client, node_controller, cluster):
+        # Define new cluster
+        cluster_id = cluster().id
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Change boot order of a random node
+        hosts = node_controller.list_nodes()
```

Comment:
looks like we need to change list nodes to return list and not dict, not sure this pr or not but sounds like this is logical